app/src/org/commcare/network/HttpUtils.java

```diff
@@ -60,22 +54,11 @@ private static String buildDomainUser(String username) {
         return username;
     }
 
-    private static String getCredential(User user, String domain) {
-        final String username;
-        if (domain != null) {
-            username = user.getUsername() + "@" + domain;
-        } else {
-            username = user.getUsername();
-        }
-        final String password = user.getCachedPwd();
-        return getCredential(username, password);
-    }
-
     private static String getCredential(String username, String password) {
         if (username == null || password == null) {
             return null;
         } else {
-            return okhttp3.Credentials.basic(username, password);
+            return okhttp3.Credentials.basic(buildDomainUser(username), buildAppPassword(password));
```

Comment:
Hey, 

So it was definitely intentional before that there was a version of this function which *didn't* try to append domain credentials. If you've audited and it's clear that we never used the domain-less version, fine to remove it, although I'd change the name of this function to something that makes it clear that it's doing double duty (IE: Creating the basic auth credential string *AND* mutating the username to attach domain data). 

I think in the context it's used, it wouldn't necessarily be unambiguous that the username you pass in here isn't the one that's going to come back out, and am worried it'll be a source of future errors.

Reply:
@ctsims Agree. Moved out the `buildDomainUser` out again since I now think it's better to not modify `username` inside the `getCredential` and let it just handle password obfuscation.